platform/mellanox/mlnx-platform-api/sonic_platform/module.py

```diff
@@ -247,3 +249,196 @@ def get_sfp(self, index):
+            return ModuleBase.MODULE_STATUS_OFFLINE
+
+    ##############################################
+    # SmartSwitch methods
+    ##############################################
+
+    def get_dpu_id(self):
+        """
+        Retrieves the DPU ID. Returns None for non-smartswitch chassis.
+        Returns:
+            An integer, indicating the DPU ID. DPU0 returns 1, DPUX returns X+1
+            Returns '0' on switch module
+        """
+        return self.dpu_id
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the cause of the previous reboot of the DPU module
+        Returns:
+            A tuple (string, string) where the first element is a string
+            containing the cause of the previous reboot. This string must
+            be one of the predefined strings in this class. If the first
+            string is "REBOOT_CAUSE_HARDWARE_OTHER", the second string can be
+            used to pass a description of the reboot cause.
+            Some more causes are appended to the existing list to handle other
+            modules such as DPUs.
+            Ex: REBOOT_CAUSE_POWER_LOSS, REBOOT_CAUSE_HOST_RESET_DPU,
+            REBOOT_CAUSE_HOST_POWERCYCLED_DPU, REBOOT_CAUSE_SW_THERMAL,
+            REBOOT_CAUSE_DPU_SELF_REBOOT
+        """
+        if utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/reset_from_main_board') == 1:
+            return self.REBOOT_CAUSE_HOST_RESET_DPU, ''
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/reset_dpu_thermal') == 1:
+            return self.REBOOT_CAUSE_SW_THERMAL, ''
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/reset_aux_pwr_or_reload') == 1:
+            return self.REBOOT_CAUSE_POWER_LOSS, ''
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/reset_pwr_off') == 1:
+            return self.REBOOT_CAUSE_DPU_SELF_REBOOT, ''
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/tpm_rst') == 1:
+            return self.REBOOT_CAUSE_HARDWARE_OTHER, 'Reset by the TPM module'
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/perst_rst') == 1:
+            return self.REBOOT_CAUSE_HARDWARE_OTHER, 'PERST# signal to ASIC'
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/phy_rst') == 1:
+            return self.REBOOT_CAUSE_HARDWARE_OTHER, 'Phy reset'
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/usbphy_rst') == 1:
+            return self.REBOOT_CAUSE_HARDWARE_OTHER, 'USB Phy reset'
+        else:
+            return self.REBOOT_CAUSE_NON_HARDWARE, ''
+
+    ##############################################
+    # Midplane methods for modular chassis
+    ##############################################
+    def get_midplane_ip(self):
```

Comment:
This should be part of the base vendor independent implementaion

Reply:
Removed